### PR TITLE
[systemd] apply sysconfig settings on launch too.

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

Honor sysconfig settings when launching with systemd service.

### Motivation

Not working in the mentioned scenario: https://github.com/DataDog/dd-agent/pull/3301

### CODE REVIEW

File: packaging/start_agent.sh [hunk 1]
```diff
@@ -5,4 +5,9 @@
 
 PATH=/opt/datadog-agent/embedded/bin:/opt/datadog-agent/bin:$PATH
 
+if [ "$DATADOG_ENABLED" = "no" ]; then
+    echo "Disabled via $AGENTSYSCONFIG. Exiting."
```
Comment (olivielpeau) [inline packaging/start_agent.sh]: `$AGENTSYSCONFIG` doesn't exist here, I'll update the PR

**Comment by olivielpeau** [comment]
@truthbk This seems to work well but the way systemd logs what our script echoes is a bit weird when `DATADOG_ENABLED` is set to `no`: 

This is what systemd logs once `service datadog-agent start` is called on an ubuntu 16.04 VM. The agent does not actually start (as expected) but the logs might be misleading:
```
$ journalctl -u datadog-agent
Aug 23 18:46:29 ubuntu-xenial systemd[1]: Starting "Datadog Agent"...
Aug 23 18:46:29 ubuntu-xenial supervisorctl[28008]: ERROR: unix:///opt/datadog-agent/run/datadog-supervisor.sock no such file (already shut down?)
Aug 23 18:46:29 ubuntu-xenial systemd[1]: Started "Datadog Agent".
```
whereas `/var/log/syslog` does contain what we echo from `start_agent.sh`:
```
Aug 23 18:46:29 ubuntu-xenial systemd[1]: Starting "Datadog Agent"...
Aug 23 18:46:29 ubuntu-xenial start_agent.sh[28007]: Disabled via DATADOG_ENABLED env var. Exiting.
Aug 23 18:46:29 ubuntu-xenial supervisorctl[28008]: ERROR: unix:///opt/datadog-agent/run/datadog-supervisor.sock no such file (already shut down?)
Aug 23 18:46:29 ubuntu-xenial systemd[1]: Started "Datadog Agent".
```

I _think_ this is related to the `forking` type of our service, which might make systemd only log what the _forked_ processes echo to stdout.

Not sure this is a blocker, but let me know what you think.

**Comment by truthbk** [comment]
The error: `ERROR: unix:///opt/datadog-agent/run/datadog-supervisor.sock no such file (already shut down?)` we see is due to `ExecStop` getting called. I think that since the script is exiting cleanly, the service is considered to have been up. Then systemd detects the process is gone, and calls `ExecStop`.

**Comment by olivielpeau** [comment]
Going to go ahead and merge this, we can fix the slightly misleading log entry later.

**Pull request merged**